lib/src/client_options.dart

```diff
@@ -69,19 +70,31 @@ class ClientOptions {
   /// Allows to enable receiving raw gateway event
   bool dispatchRawShardEvent;
 
+  /// The [RetryOptions] to use when a shard fails to connect to the gateway.
+  RetryOptions shardReconnectOptions;
+
+  /// The [RetryOptions] to use when a HTTP request fails.
+  ///
+  /// Note that this will not retry requests that fail because of their HTTP response code (e.g  a 4xx response) but rather requests that fail due to native
+  /// errors (e.g failed host lookup) which can occur if there is no internet.
+  RetryOptions httpRetryOptions;
+
   /// Makes a new `ClientOptions` object.
-  ClientOptions(
-      {this.allowedMentions,
-      this.shardCount,
-      this.messageCacheSize = 100,
-      this.largeThreshold = 50,
-      this.compressedGatewayPayloads = true,
-      this.guildSubscriptions = true,
-      this.initialPresence,
-      this.shutdownHook,
-      this.shutdownShardHook,
-      this.dispatchRawShardEvent = false,
-      this.shardIds});
+  ClientOptions({
+    this.allowedMentions,
+    this.shardCount,
+    this.messageCacheSize = 100,
+    this.largeThreshold = 50,
+    this.compressedGatewayPayloads = true,
+    this.guildSubscriptions = true,
+    this.initialPresence,
+    this.shutdownHook,
+    this.shutdownShardHook,
+    this.dispatchRawShardEvent = false,
+    this.shardIds,
+    this.shardReconnectOptions = const RetryOptions(maxAttempts: 50), // 50 attempts won't ever reasonably be reached
+    this.httpRetryOptions = const RetryOptions(maxAttempts: 50),
```

Comment:
i think we should lower max attempts here. I think max of ~10 should be enough and other than that should require manual intervention